File: py/problem_6.py
#!/usr/bin/env python

def square(x: int) -> int:
    return x * x


def square_of_sum(x: int) -> int:
    return square(sum(range(1, x + 1)))


def sum_of_squares(x: int) -> int:
    return sum(map(square, range(1, x + 1)))


print(square_of_sum(100) - sum_of_squares(100))

File: dart/problem_6.dart
#!/usr/bin/env dart

import 'iterables.dart';
import 'package:collection/collection.dart';

int square(int x) => x * x;
int squareOfSum(int x) => square(range(1, x).sum);
int sumOfSquares(int x) => range(1, x).map(square).sum;

void main() {
  print(squareOfSum(100) - sumOfSquares(100));
}
